### DISCUSSION TITLE
How to run the pre-commit hook only for changed files

### DISCUSSION THREAD

**Discussion by henningBunk** [detekt/detekt#3632] [Q&A] [answered]
I like the idea of the pre-commit hook but running detekt for all files takes a while and isn't really necessary. I read [here](https://github.com/detekt/detekt/discussions/3303#discussioncomment-225557) that it is possible to run the hook only for the changed files and this should speed up the process.

How would I do this with gradle? Or is only possible/more convenient using the CLI?

**Comment by BraisGabin** [comment]
I know little about gradle to be honest. But the gradle plugin has a `input` parameter that could be use exactly for that. But if you use type solving I'm not sure how much time you will save.

**Comment by schalkms** [comment, marked as answer]
This has been answered in the following threads. Please have a look here:

1. https://github.com/detekt/detekt/issues/3078
2. https://github.com/detekt/detekt/issues/2320
3. https://github.com/detekt/detekt/issues/1792
